src/features/eighteenBirthday/pages/components/EighteenBirthdayCard.tsx

```diff
@@ -2,17 +2,29 @@ import { t } from '@lingui/macro'
 import { useNavigation } from '@react-navigation/native'
 import React from 'react'
 
-import { NavigateToHomeWithoutModalOptions } from 'features/navigation/helpers'
+import { useGetIdCheckToken } from 'features/auth/api'
+import { useUserProfileInfo } from 'features/home/api'
 import { UseNavigationType } from 'features/navigation/RootNavigator'
 import { _ } from 'libs/i18n'
 import TutorialPassLogo from 'ui/animations/eighteen_birthday.json'
 import { AchievementCardKeyProps, GenericAchievementCard } from 'ui/components/achievements'
+import { useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
 
 export function EighteenBirthdayCard(props: AchievementCardKeyProps) {
   const { navigate } = useNavigation<UseNavigationType>()
+  const { data: idCheck } = useGetIdCheckToken(true)
+  const { data: profile } = useUserProfileInfo()
+  const { displayInfosSnackBar } = useSnackBarContext()
 
   function onButtonPress() {
-    navigate('Home', NavigateToHomeWithoutModalOptions)
+    if (profile && idCheck) {
+      navigate('IdCheck', { email: profile.email, licenceToken: idCheck.token || '' })
```

Comment:
On ne peut pas aller sur IdCheck sans token. Il faut gérer le cas où le token est possiblement falsy 
```suggestion
      navigate('IdCheck', { email: profile.email, licenceToken: idCheck.token })
```

Reply:
`if (profile && idCheck?.token) {}` ?